src/posit/connect/external/databricks.py

```diff
@@ -5,73 +5,89 @@
 from ..client import Client
 from ..oauth import OAuthIntegration
 
-HeaderFactory = Callable[[], Dict[str, str]]
-
-
-# https://github.com/databricks/databricks-sdk-py/blob/v0.20.0/databricks/sdk/credentials_provider.py
-# https://github.com/databricks/databricks-sql-python/blob/v3.1.0/src/databricks/sql/auth/authenticators.py
-# In order to keep compatibility with the Databricks SDK
-class CredentialsProvider(abc.ABC):
-    """Protocol Databricks authentication.
-
-    A call-side interface for the Databricks Rest API.
-    """
-
+"""
+NOTE: These APIs are provided as a convenience and are subject to breaking changes:
+https://github.com/databricks/databricks-sdk-py#interface-stability
+"""
+
+# The Databricks SDK CredentialsProvider == Databricks SQL HeaderFactory
+CredentialsProvider = Callable[[], Dict[str, str]]
+
+# In order to keep compatibility with the Databricks SQL/SDK client libraries:
+# https://github.com/databricks/databricks-sql-python/blob/v3.3.0/src/databricks/sql/auth/authenticators.py#L19-L33
+# https://github.com/databricks/databricks-sdk-py/blob/v0.29.0/databricks/sdk/credentials_provider.py#L44-L54
+class CredentialsStrategy(abc.ABC):
     @abc.abstractmethod
     def auth_type(self) -> str:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def __call__(self, *args, **kwargs) -> HeaderFactory:
+    def __call__(self, *args, **kwargs) -> CredentialsProvider:
         raise NotImplementedError
 
 
-class PositOAuthIntegrationCredentialsProvider(CredentialsProvider):
+# Use this environment variable to determine if the
+# client SDK was initialized from a piece of content running on a Connect server.
+def _is_local() -> bool:
+    return not os.getenv("RSTUDIO_PRODUCT") == "CONNECT"
+
+
+class PositOAuthIntegrationCredentialsProvider:
     def __init__(self, posit_oauth: OAuthIntegration, user_session_token: str):
         self.posit_oauth = posit_oauth
         self.user_session_token = user_session_token
 
-    def auth_type(self) -> str:
-        return "posit-oauth-integration"
-
-    def __call__(self, *args, **kwargs) -> HeaderFactory:
-        def inner() -> Dict[str, str]:
-            access_token = self.posit_oauth.get_credentials(
-                self.user_session_token
-            )["access_token"]
-            return {"Authorization": f"Bearer {access_token}"}
-
-        return inner
+    def __call__(self) -> Dict[str, str]:
+        access_token = self.posit_oauth.get_credentials(
+            self.user_session_token
+        )["access_token"]
+        return {"Authorization": f"Bearer {access_token}"}
 
 
-# Use this environment variable to determine if the
-# client SDK was initialized from a piece of content running on a Connect server.
-def is_local() -> bool:
-    return not os.getenv("RSTUDIO_PRODUCT") == "CONNECT"
+class PositOAuthIntegrationCredentialsStrategy(CredentialsStrategy):
```

Comment:
I know this is superficial, but do we have options for making this class name shorter? As a user, giant names like this scare me and make me assume that the code is super complex. 

Would `PositCredentialsStrategy` work?

Reply:
Same here! That or `ConnectCredentialsStrategy`.

Reply:
Just another option in the merry-go-round of word permutations - because this is already under the `posit/connect` directory, would it be more informative to call it an `OAuthCredentialsStrategy`? Then we could have other credentials strategies that also work for posit products / connect but we don't run into naming collisions.

Reply:
I opted to go with `PositCredentialsStrategy` to avoid a naming conflict with the databricks sdk's `OAuthCredentialsStrategy`.  I think this is a good balance of brevity and descriptive-ness since we are still under the `posit/connect` namespace 

https://github.com/databricks/databricks-sdk-py/blob/7d22b4d3727478f0f5dbeb34b7f6fc17a03e31b7/databricks/sdk/credentials_provider.py#L57